src/features/home/pages/useHomeModules.ts

```diff
@@ -37,49 +37,53 @@ export const useHomeModules = (
   offerModules: Array<Offers | OffersWithCover>
 ): HomeModuleResponse => {
   const { position } = useGeolocation()
-  const [homeModules, setHomeModules] = useState<HomeModuleResponse>({})
+  const homeModules: HomeModuleResponse = {}
   const { enabled, fetchMultipleHits, filterHits, transformHits } = useFetchMultipleHits()
 
   const queries = useQueries(
-    offerModules.map(({ search, moduleId }) => {
-      const parsedParameters = search
-        .map((parameters) => parseSearchParameters({ geolocation: position, parameters }))
-        .filter(isParsedParameter)
+    offerModules.map(
+      ({ search, moduleId }: { search: SearchParametersFields[]; moduleId: string }) => {
+        const parsedParameters = search
+          .map((parameters) => parseSearchParameters({ geolocation: position, parameters }))
+          .filter(isParsedParameter)
 
-      const fetchModule = async () => {
-        const response = await fetchMultipleHits(parsedParameters)
-        return { moduleId: moduleId, ...response }
+        const fetchModule = async () => {
+          const response = await fetchMultipleHits(parsedParameters)
+          return { moduleId: moduleId, ...response }
+        }
+
+        return {
+          queryKey: [QueryKeys.HOME_MODULE, moduleId],
+          queryFn: fetchModule,
+          enabled,
+          notifyOnChangeProps: ['data'],
+        }
       }
+    )
+  )
 
-      return {
-        queryKey: [QueryKeys.HOME_MODULE, moduleId],
-        queryFn: fetchModule,
-        onSuccess: (response) => {
-          let hits: SearchHit[] = []
-          let nbHits = 0
+  queries.forEach((query) => {
+    let hits: SearchHit[] = []
+    let nbHits = 0
 
-          if (isMultipleAlgoliaHit(response)) {
-            hits = flatten(response.results.map(({ hits }) => hits))
-            nbHits = response.results.reduce((prev, curr) => prev + curr.nbHits, 0)
-          } else if (isMultipleSearchHit(response)) {
-            hits = response.hits
-            nbHits = response.nbHits
-          } else {
-            return
-          }
+    if (!query.isSuccess) return
 
-          setHomeModules((prevHomeModules) => ({
-            ...prevHomeModules,
-            [response.moduleId]: {
-              hits: uniqBy(hits.filter(filterHits).map(transformHits), 'objectID'),
-              nbHits,
-            },
-          }))
-        },
-        enabled,
-      }
-    })
-  )
+    const data = query.data
+    if (isMultipleAlgoliaHit(data)) {
+      hits = flatten(data.results.map(({ hits }) => hits))
+      nbHits = data.results.reduce((prev, curr) => prev + curr.nbHits, 0)
```

Comment:
Note pour Algolia uniquement. Si tu fais le `flatten` et le `reduce` là, ils seront effectués à chaque appel de ce hook (vu que `homeModules` est réinitialisé à chaque fois).

Pour palier à ça, tu peux modifier la fonction `fetchModule` comme l'extrait que je t'ai passé, comme ça c'est appelé uniquement quand on a la réponse, ou tu peux passer homeModules en référence, et skipper la phase de calcul:

```js
    if (data?.moduleId in homeModules) return
```

Reply:
Ca ne va pas poser pb si on a envie de mettre à jour les données récupérées ?

Reply:
on a pas besoin de mettre à jour les données. Et en plus ça c'est uniquement dans le cas où on fait ces calculs dans le `forEach`. Pas besoin `if (data?.moduleId in homeModules) return` si on fait les calculs plus haut